safe_transaction_service/history/tests/test_models.py

```diff
@@ -227,6 +227,30 @@ class TestEthereumTx(TestCase):
 
 
 class TestTokenTransfer(TestCase):
+    def test_transfer_to_erc721(self):
+        erc20_transfer = ERC20TransferFactory()
+        self.assertEqual(ERC721Transfer.objects.count(), 0)
+        erc20_transfer.to_erc721_transfer().save()
+        self.assertEqual(ERC721Transfer.objects.count(), 1)
+        erc721_transfer = ERC721Transfer.objects.get()
+        self.assertEqual(erc721_transfer.ethereum_tx_id, erc20_transfer.ethereum_tx_id)
+        self.assertEqual(erc721_transfer.address, erc20_transfer.address)
+        self.assertEqual(erc721_transfer.log_index, erc20_transfer.log_index)
+        self.assertEqual(erc721_transfer.to, erc20_transfer.to)
+        self.assertEqual(erc721_transfer.token_id, erc20_transfer.value)
```

Comment:
Ah I see, sometimes ERC721 transfers get indexed as ERC20 events?

Reply:
Yes, as the event should be `Transfer(indexed address from, indexed address to, indexed uint256 tokenId)`, but sometimes it is `Transfer(indexed address from, indexed address to, uint256 tokenId)` matching the one for `ERC20` and making it impossible to tell apart